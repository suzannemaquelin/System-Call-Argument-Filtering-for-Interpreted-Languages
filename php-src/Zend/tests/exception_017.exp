Exception: Cannot call abstract method C::foo() in %sexception_017.php on line %d

Exception: Argument 1 passed to foo() must be callable, string given, called in %sexception_017.php on line %d
Exception: Cannot call abstract method C::foo()

Fatal error: Uncaught Error: Cannot call abstract method C::foo() in %sexception_017.php:%d
Stack trace:
#0 {main}
  thrown in %sexception_017.php on line %d