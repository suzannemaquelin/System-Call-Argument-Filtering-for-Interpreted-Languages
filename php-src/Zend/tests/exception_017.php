<?php
abstract class C {
	abstract static function foo();
}

function foo(callable $x) {
}

try {
	C::foo();
} catch (Error $e) {
	echo "\nException: " . $e->getMessage() . " in " , $e->getFile() . " on line " . $e->getLine() . "\n";
}

try {
	foo("C::foo");
} catch (Error $e) {
	echo "\n";
	do {
		echo "Exception: " . $e->getMessage() . "\n";
		$e = $e->getPrevious();
	} while ($e instanceof Error);
}

C::foo();
?>
