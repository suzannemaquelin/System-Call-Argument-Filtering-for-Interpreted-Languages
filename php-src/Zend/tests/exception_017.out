Failed loading /opt/php-7.1/lib/php/extensions/debug-non-zts-20160303/xdebug.so:  /opt/php-7.1/lib/php/extensions/debug-non-zts-20160303/xdebug.so: cannot open shared object file: No such file or directory

Exception: Cannot call abstract method C::foo() in /root/php-src/Zend/tests/exception_017.php on line 10

Exception: Argument 1 passed to foo() must be callable, string given, called in /root/php-src/Zend/tests/exception_017.php on line 16
Exception: Cannot call abstract method C::foo()

Fatal error: Uncaught Error: Cannot call abstract method C::foo() in /root/php-src/Zend/tests/exception_017.php:25
Stack trace:
#0 {main}
  thrown in /root/php-src/Zend/tests/exception_017.php on line 25